checker/jdk/nullness/src/org/w3c/dom/CharacterData.java

```diff
@@ -0,0 +1,185 @@
+    /**
+     * The character data of the node that implements this interface. The DOM
+     * implementation may not put arbitrary limits on the amount of data
+     * that may be stored in a <code>CharacterData</code> node. However,
+     * implementation limits may mean that the entirety of a node's data may
+     * not fit into a single <code>DOMString</code>. In such cases, the user
+     * may call <code>substringData</code> to retrieve the data in
+     * appropriately sized pieces.
+     * @exception DOMException
+     *   NO_MODIFICATION_ALLOWED_ERR: Raised when the node is readonly.
+     */
+    public void setData(@Nullable String data)
+                            throws DOMException;
+
+    /**
+     * The number of 16-bit units that are available through <code>data</code>
+     * and the <code>substringData</code> method below. This may have the
+     * value zero, i.e., <code>CharacterData</code> nodes may be empty.
+     */
+    @Pure public int getLength();
+
+    /**
+     * Extracts a range of data from the node.
+     * @param offset Start offset of substring to extract.
+     * @param count The number of 16-bit units to extract.
+     * @return The specified substring. If the sum of <code>offset</code> and
+     *   <code>count</code> exceeds the <code>length</code>, then all 16-bit
+     *   units to the end of the data are returned.
+     * @exception DOMException
+     *   INDEX_SIZE_ERR: Raised if the specified <code>offset</code> is
+     *   negative or greater than the number of 16-bit units in
+     *   <code>data</code>, or if the specified <code>count</code> is
+     *   negative.
+     *   <br>DOMSTRING_SIZE_ERR: Raised if the specified range of text does
+     *   not fit into a <code>DOMString</code>.
+     */
+    public String substringData(int offset,
+                                int count)
+                                throws DOMException;
+
+    /**
+     * Append the string to the end of the character data of the node. Upon
+     * success, <code>data</code> provides access to the concatenation of
+     * <code>data</code> and the <code>DOMString</code> specified.
+     * @param arg The <code>DOMString</code> to append.
+     * @exception DOMException
+     *   NO_MODIFICATION_ALLOWED_ERR: Raised if this node is readonly.
+     */
+    public void appendData(String arg)
+                           throws DOMException;
+
+    /**
+     * Insert a string at the specified 16-bit unit offset.
+     * @param offset The character offset at which to insert.
+     * @param arg The <code>DOMString</code> to insert.
+     * @exception DOMException
+     *   INDEX_SIZE_ERR: Raised if the specified <code>offset</code> is
+     *   negative or greater than the number of 16-bit units in
+     *   <code>data</code>.
+     *   <br>NO_MODIFICATION_ALLOWED_ERR: Raised if this node is readonly.
+     */
+    public void insertData(int offset,
+                           @Nullable String arg)
```

Comment:
CharacterData is sub class of Node, both Node value and  DOMString can be null. parameter in appendData function should be annotated.